src/app/utils/field-validation/validators/dateValidator.ts

```diff
@@ -106,6 +110,77 @@ const makeDateValidator: DateValidatorConstructor = (dateField) => {
   }
 }
 
+/**
+ * Returns the number representation of a day of the week
+ */
+const convertInvalidDayToNumber = (invalidDay: InvalidDaysOptions): number => {
+  switch (invalidDay) {
+    case InvalidDaysOptions.Sunday:
+      return 0
+    case InvalidDaysOptions.Monday:
+      return 1
+    case InvalidDaysOptions.Tuesday:
+      return 2
+    case InvalidDaysOptions.Wednesday:
+      return 3
+    case InvalidDaysOptions.Thursday:
+      return 4
+    case InvalidDaysOptions.Friday:
+      return 5
+    case InvalidDaysOptions.Saturday:
+      return 6
+    default:
+      return -1
+  }
+}
+
+/**
+ * Convert the days of the week in the invalidDays array
+ * to a number array representing the number representation
+ * of the corresponding day of the week
+ */
+const convertInvalidDaysOfTheWeekToNumberSet = (
+  invalidDays: InvalidDaysOptions[],
+): Set<number> => {
+  if (invalidDays.length === 0) {
+    return new Set()
+  }
+
+  const invaliDaysNumberArray = invalidDays.map((invalidDay) =>
+    convertInvalidDayToNumber(invalidDay),
+  )
```

Comment:
also typo in `invaliDaysNumberArray`

nit: then the constant can be used the same way:

```tsx
return new Set(invalidDays.map(invalidDay => INVALID_DAY_MAP[invalidDay] ?? -1))
```